src/Orleans/Configuration/ClientConfiguration.cs

```diff
@@ -137,6 +138,11 @@ public enum GatewayProviderType
         public bool StatisticsWriteLogStatisticsToTable { get; set; }
         public StatisticsLevel StatisticsCollectionLevel { get; set; }
 
+        /// <summary>
+        /// Gets or sets the service configuration delegate.
+        /// </summary>
+        public Func<IServiceCollection, IServiceProvider> ServiceConfigurator { get; set; } = services => services.BuildServiceProvider();
```

Comment:
I should remove it from this PR and we can talk about introducing it later. It's a good hook to have, and I consider it configuration since it's used only as part of the configuration phase and is the same signature as our Startup.ConfigureServices method (& same as ASP.NET), but it's not serializable.